packages/context/src/lib/controllers/context-provider.ts

```diff
@@ -94,13 +94,16 @@ export class ContextProvider<
     ev: ContextRequestEvent<Context<unknown, unknown>>
   ): void => {
     // Only call the callback if the context matches.
+    if (ev.context !== this.context) {
+      return;
+    }
     // Also, in case an element is a consumer AND a provider
     // of the same context, we want to avoid the element to self-register.
     // The check on composedPath (as opposed to ev.target) is to cover cases
     // where the consumer is in the shadowDom of the provider (in which case,
     // event.target === this.host because of event retargeting).
     const consumerHost = ev.composedPath()[0] as Element;
-    if (ev.context !== this.context || consumerHost === this.host) {
+    if (consumerHost === this.host) {
```

Comment:
When the consumer is in the light DOM of the provider, I think we can further reduce the calls to `ev.composedPath()` by checking target first:

```suggestion
    if (event.target === this.host && ev.composedPath()[0] === this.host) {
```

Same below.


We could also avoid `composedPath()` altogether if we added another property to the ContextRequestEvent, like `contextTarget`. We'd set this to the dispatching host before dispatch and check it here. If the property doesn't exist, then we fallback to composed path to be compatible with the context protocol.

If that's really beneficial, we can bring `.contextTarget` to the protocol, and also open a standards issue for a property like `composedTarget` for these cases when you really don't need to whole path.

Reply:
> When the consumer is in the light DOM of the provider, I think we can further reduce the calls to ev.composedPath() by checking target first

Good spot, applied the change to both places :+1:

Reply:
If you're up to it I do think it'd be better to put a `contextTarget` property on ContextEvent. I suspect that a majority of context use is with consumers in the shadow root of the provider, so this small addition check here won't help them that much, while `contextTarget` could eliminate the composed path check for all Lit-to-Lit context scenarios.